architectures/BDR-Always-ON/commands/upgrade.yml

```diff
@@ -98,6 +103,22 @@
   # (e.g., primary instances) or not (e.g., logical standby instances).
   # We don't do anything for non-Postgres instances at the moment.
 
+  - name: Check to perform major upgrade for Postgres
+    set_fact:
+      postgres_major_version_upgrade: "{{ cluster_facts.postgres_version is version(postgres_version, 'lt') }}"
+
+  - name: Stop/upgrade Postgres on {{ inventory_hostname }}
+    block:
+    - include_role: name=postgres/bdr tasks_from=fence.yml
+      when: >
+        'readonly' not in role and 'witness' not in role
+    - include_role: name=postgres/update
```

Comment:
yes, config is handled in roles/postgres/pkg/tasks/upgrade.yml. I can adjust if required. This block was created separately because of the fencing requirement as well as readability in this PR.